app/Coupon.php

```diff
@@ -21,7 +23,19 @@ public function admin(){
     public function buyers(){
         return $this->hasMany(Buyer::class);
     }
- 
+    /**
+     * fucntion to check if the lifeTime for coupon is available or no
+     */
+    public  function check_lifeTime_for_coupon($coupon){
+        $lifeTime = Carbon::parse($coupon->lifetime);
+        if($lifeTime > now()){
+            return TRUE;
+        }else{
+            $coupon->validate_state = 0;
+            $coupon->save();
+            return FALSE;
+        }
+    }
```

Comment:
change TRUE AND FALSE TO true and false.

Change $coupon->lifetime => $this->lifetime